node/narwhal/src/helpers/storage.rs

```diff
@@ -534,6 +569,131 @@ impl<N: Network> Storage<N> {
     }
 }
 
+impl<N: Network> Storage<N> {
+    /// Syncs the storage with the ledger.
+    pub fn sync_storage_with_ledger(&self) -> Result<()> {
+        // Retrieve the latest block in the ledger.
+        let latest_block = self.ledger.latest_block();
+        // Sync the storage with the block.
+        self.sync_storage_with_block(latest_block)
+    }
+
+    /// Syncs the storage with the block.
+    pub fn sync_storage_with_block(&self, block: Block<N>) -> Result<()> {
+        // Retrieve the block height.
+        let block_height = block.height();
+        // Determine the earliest height, conservatively set to the block height minus the max GC rounds.
+        // By virtue of the BFT protocol, we can guarantee that all GC range blocks will be loaded.
+        let gc_height = block_height.saturating_sub(u32::try_from(self.max_gc_rounds)?);
```

Comment:
There could be many rounds between blocks, so this is treating `max_gc_rounds` more as `max_gc_block_heights`.